### PR TITLE
Check if the available transfer quota is enough

### PR DESCRIPTION
**Pull request by javiserrano**

### CODE REVIEW

**Comment by javiergm1983** [comment]
trigger compilation

File: include/mega/megaclient.h [hunk 1]
```diff
@@ -277,6 +277,9 @@ class MEGA_API MegaClient
     // retrieve user details
     void getaccountdetails(AccountDetails*, bool, bool, bool, bool, bool, bool);
 
+    // check if the available bandwidth quota is enough to transfer an amount of bytes
+    void querybandwidthquota(m_off_t size);
+
```
Comment (sergiohs84) [inline include/mega/megaclient.h]: Maybe we should refactorize all these to be _transfer quota_, which is the actual name of the quota and how it's displayed to users.

File: src/commands.cpp [hunk 2]
```diff
@@ -3229,6 +3229,26 @@ void CommandGetUserQuota::procresult()
...
+    tag = client->reqtag;
+}
+
+void CommandQueryBandwidthQuota::procresult()
+{
+    if (!client->json.isnumeric())
+    {
+        LOG_err << "Unexpected response: " << client->json.pos;
+        client->json.storeobject();
+        return client->app->querybandwidthquota_result(0);
```
Comment (sergiohs84) [inline src/commands.cpp]: Shouldn't we return `API_EINTERNAL` in this case??

File: src/megaapi_impl.cpp [hunk 3]
```diff
@@ -10197,6 +10206,16 @@ void MegaApiImpl::account_details(AccountDetails*, error e)
     fireOnRequestFinish(request, megaError);
 }
 
+void MegaApiImpl::querybandwidthquota_result(int code)
+{
+    if(requestMap.find(client->restag) == requestMap.end()) return;
+    MegaRequestPrivate* request = requestMap.at(client->restag);
+    if(!request || (request->getType() != MegaRequest::TYPE_QUERY_BANDWIDTH_QUOTA)) return;
+
```
Comment (sergiohs84) [inline src/megaapi_impl.cpp]: If we return in the `Command::procresult()` an `API_EINTERNAL` (or any other error code), I think we should not set the flag in the request unless the code is `API_OK`

File: bindings/wp8/MegaSDK.cpp [hunk 4]
```diff
@@ -1843,6 +1843,16 @@ void MegaSDK::getExtendedAccountDetails(bool sessions, bool purchases, bool tran
     megaApi->getExtendedAccountDetails(sessions, purchases, transactions);
 }
 
+void MegaSDK::queryBandwidthQuota(int64 size, MRequestListenerInterface^ listener)
+{
+    megaApi->queryBandwidthQuota(size, createDelegateMRequestListener(listener));
```
Comment (sergiohs84) [inline bindings/wp8/MegaSDK.cpp]: This MegaApi function has been refactored to `queryTransferQuota()`. You need to change this call in the binding.

File: bindings/wp8/MegaSDK.cpp [hunk 5]
```diff
@@ -1843,6 +1843,16 @@ void MegaSDK::getExtendedAccountDetails(bool sessions, bool purchases, bool tran
...
+}
+
+void MegaSDK::queryBandwidthQuota(int64 size)
+{
+    megaApi->queryBandwidthQuota(size);
```
Comment (sergiohs84) [inline bindings/wp8/MegaSDK.cpp]: Same, need to refactor.

**Comment by javiserrano** [comment]
I think that it's OK now

**Pull request merged**